### PR TITLE
Fixes `kubectl cp` copy to local file-mode missing

### PR DESCRIPTION
**Pull request by wzshiming**
#### What type of PR is this?
/kind bug
/sig cli

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Fixes copy to local file-mode missing
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

**Comment by eddiezane** [comment]
@wzshiming thank you for the PR.

Could you please explain with an example why this change is necessary?

We'd also need a test for this to catch future regressions.

**Comment by wzshiming** [comment]
Copy files from the container to local and want to keep the file mode

**Comment by wzshiming** [comment]
@eddiezane PTAL

File: staging/src/k8s.io/kubectl/pkg/cmd/cp/cp_test.go [hunk 1]
```diff
@@ -411,7 +412,7 @@ func TestTarUntarWrongPrefix(t *testing.T) {
 	if err := os.MkdirAll(path.Dir(filepath), 0755); err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
-	createTmpFile(t, filepath, "sample data")
+	createTmpFile(t, filepath, "sample data", 0666)
```
Comment (dougsland) [inline staging/src/k8s.io/kubectl/pkg/cmd/cp/cp_test.go:430]: How about adding a const for the perm so you can check later in the code if the perm is correct?
Additional to that, I would go with 0664 even in tests. :)

**Comment by wzshiming** [comment]
Sorry for the late reply, I seem to miss this PR

**Pull request closed without merging**